processors/base/helpers/__init__.py

```diff
@@ -266,3 +279,47 @@ def safe_prepend(prepend_string, string):
         string = '%s%s' % (prepend_string, string)
 
     return string
+
+
+def get_canonical_location_name(location):
+    """Find the canonical location name according to the
+    passed entry
+
+    Args:
+        location (str): the location to be normalized
+    """
+
+    if location is None:
+        return None
+
+    # Extracted from: https://github.com/datasets/country-codes/blob/master/data/country-codes.csv
+    CSV_PATH = os.path.join(os.path.dirname(__file__), 'data/countries.csv')
+    DISTANCE_SCORER = fuzzywuzzy.fuzz.token_sort_ratio
+
+    clean_string = lambda u: re.sub(u"[^\w\d'\s]+", '', u).lower()
+    try:
+        logger.debug('Location "%s" normalized using ISO-3166 standard', location)
```

Comment:
You don't know if it was normalized here. The following line can still throw an exception. It's also good to log the normalized value, so you can change this to:

```python
normalized_location = iso3166.countries.get(cleaned_location).name
logger.debug('Location "%s" normalized as "%s"', cleaned_location, normalized_location)
return normalized_location
```

This also mean moving the `cleaned_location = clean_string(location)` from line 312 to before here, possibly before the `try` block.